bot.py

```diff
@@ -296,6 +304,29 @@ async def create_or_get_voice_client(self, message):
 
         return self.voice_client
 
+    @tasks.loop(seconds=2)
+    async def disconnect_checker(self):
+        """
+        Checks every few seconds whether the voice_client is playing something in a
+        voice channel. If it's not then the bot will be automatically disconnected.
+        """
+        if self.voice_client and not self.voice_client.is_playing():
```

Comment:
```suggestion
        if not self.voice_client or self.voice_client.is_playing():
            return
        # ...rest of code
```

You could make the code slightly more readable by reducing nesting.

Reply:
nice!